src/components/OrderHistory/OrderHistoryList.tsx

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from "react";
+import { NavLink } from "react-router-dom";
+import { navKeys } from "../../routes/routes";
+
+import { fetchOrderHistory } from "../../common/service/common-service";
+import OrderItem from "./OrderItem";
+
+import { Button } from "../../common/styles/componentsStyles";
+import { StyledOrderHistoryList } from "./OrderHistoryListStyles";
+
+import { OrderInterface } from "../../common/types/common.types";
+
+const OrderHistoryList = () => {
+  const [orders, setOrders] = useState<OrderInterface[] | null>([]);
+
+  useEffect(() => {
+    const fetchOrders = async () => {
+      const data = await fetchOrderHistory();
+      setOrders(data);
+    };
+    fetchOrders();
+  }, []);
+
+  //guard clause
+  if (!orders)
+    return <p>No orders. Please make your fisrt order to see history!</p>;
+
+  return (
+    <StyledOrderHistoryList>
+      {orders &&
+        orders.map((item) => (
+          <OrderItem
+            orderedCandies={item.orderedCandies}
+            address={item.address}
+            date={item.date}
+            // TODO:
+            // key={item.date}
+            key={Math.random.toString()}
```

Comment:
the Math.random.toString() key is not "stable". The best approach is to use item.id (as we fetch items from the server, they all should have this field (or something else that's unique).